### DISCUSSION TITLE
How to simply import and use custom global components in mdx?

### DISCUSSION THREAD

**Discussion by ayangweb** [web-infra-dev/rspress#1500] [Q&A] [answered]
大佬们好，mdx 如何简单的导入使用自定义全局组件？

我看官网写的是如下图所示：

<img width="1541" alt="iShot_2024-10-18_10 34 46" src="https://github.com/user-attachments/assets/200ff434-b352-4950-bf51-0bef04c0a97b">

但是如果在 mdx 不导入的话就报错：

<img width="1541" alt="image" src="https://github.com/user-attachments/assets/963bec05-722b-48a8-a47b-33f71b8fdd69">

但是手动导入的路径又过于复杂：

<img width="1542" alt="image" src="https://github.com/user-attachments/assets/333d79c6-ec08-4817-a3f3-f767e601bce9">

图片也没办法正常显示（路径和上面显示出来的图片路径一致）：

<img width="1541" alt="iShot_2024-10-18_10 37 20" src="https://github.com/user-attachments/assets/4954dd82-68c2-425a-90ec-3b6491e89d00">

<img width="1541" alt="image" src="https://github.com/user-attachments/assets/6556dace-cc33-4b50-94d7-962c5b249aab">

**Comment by SoonIter** [comment, marked as answer]
这个问题是把
全局UI组件和 mdx 自定义组件混淆了
你这篇文档说的是每个页面都会有的组件，比如 ScrollToTop

mdx的配置里有一个globalUIComponent，声明后是可以在mdx中直接使用 <Image />

https://rspress.dev/zh/api/config/config-build#markdownglobalcomponents

**Reply by SoonIter** [reply]
另外建议问题用英语哈，方便检索

**Comment by ayangweb** [comment]
好的，谢谢🌹注册组件的问题解决了，图片路径的问题我下来再研究研究，图片路径没有被解析！

<img width="1541" alt="image" src="https://github.com/user-attachments/assets/b2e544bf-525d-4ef0-a4ab-65a749049a5a">
